index.js

```diff
@@ -10,26 +10,48 @@ Geoservices.prototype.featureServer = function (req, res) {
   })
 }
 
+/**
+ * Handler for the $namepace/rest/info route. Inspects model for authentation info and passes any on to the
+ * FeatureServer handler
+ * @param {*} req
+ * @param {*} res
+ */
 Geoservices.prototype.featureServerRestInfo = function (req, res) {
-  FeatureServer.route(req, res)
+  // Inspect model for an "serviceAuthenticationSpecification" function; if undefined create a dummy function that returns an empty object
+  let authSpecs = this.model.serviceAuthenticationSpecification || function () { return {} }
+  FeatureServer.route(req, res, authSpecs(`${req.protocol}://${req.headers.host}`))
+}
+
+/**
+ * Handler for $namespace/authenticate route. Passes request and response object to the model's "authenticate" function
+ * @param {*} req
+ * @param {*} res
+ */
+Geoservices.prototype.authenticate = function (req, res) {
+  this.model.authenticate(req, res)
 }
 
 /**
  * Collection of route objects that define geoservices
- * 
- * These routes are bound to the Koop API for each provider. Note that FeatureServer, 
- * FeatureServer/layers, FeatureServer/:layer, and FeatureServer/:layer/:method are found 
+ *
+ * These routes are bound to the Koop API for each provider. Note that FeatureServer,
+ * FeatureServer/layers, FeatureServer/:layer, and FeatureServer/:layer/:method are found
  * in the collection with and without the "$namespace/rest/services/$providerParams" prefix.
- * These prefixed routes have been added due to some clients requiring the "rest/services" 
- * URL fragment in geoservices routes. The $namespace and $providerParams are placeholders 
- * that koop-core replaces with provider specific settings.  
+ * These prefixed routes have been added due to some clients requiring the "rest/services"
+ * URL fragment in geoservices routes. The $namespace and $providerParams are placeholders
+ * that koop-core replaces with provider-specific settings.
  */
 Geoservices.routes = [
   {
     path: '$namespace/rest/info',
     methods: ['get', 'post'],
     handler: 'featureServerRestInfo'
   },
+  {
+    path: '$namespace/authenticate',
```

Comment:
is this route called by the http client at any point?

Reply:
yes, frequently.  Anytime the client gets a 499 it appears to call this route.

Reply:
well then. I had no idea that was even a thing!

Reply:
Yeah. In ArcGIS server, the route is called `generateToken` rather than `authenticate` as we have done here. The name doesn't matter, because the clients get the route URL from the `rest/info` service. But yeah, this route is called when the user first submits their creds on AGOL prompt, as well as on receiving 499 when the originally-issued token expires.  Do you still think it should be here, rather than a custom-route on a provider?

Reply:
Updated in 99735831e6151e6cbaf7834947faf8a2092e7388.